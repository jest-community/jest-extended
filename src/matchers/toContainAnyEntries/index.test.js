import matcher from './';

expect.extend(matcher);

const data = { a: 'foo', b: 'bar', c: 'baz' };

describe('.toContainAnyEntries', () => {
  it('passes when given object contains atleast one of the given entries', () => {
    expect(data).toContainAnyEntries([['a', 'qux'], ['a', 'foo']]);
  });

  it('fails when given object does not contain any of the given entries', () => {
    expect(() => expect(data).toContainAnyEntries([['a', 'qux'], ['b', 'foo']])).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toContainAnyEntries', () => {
  it('passes when given object does not contain any of the given entries', () => {
    expect(data).not.toContainAnyEntries([['a', 'qux'], ['b', 'foo']]);
  });

  it('fails when given object contains atleast one of the given entries', () => {
    expect(() => expect(data).not.toContainAnyEntries([['a', 'qux'], ['a', 'foo']])).toThrowErrorMatchingSnapshot();
  });
});
