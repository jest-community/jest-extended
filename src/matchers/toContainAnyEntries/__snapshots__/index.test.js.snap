// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toContainAnyEntries fails when given object contains atleast one of the given entries 1`] = `
"expect(received).not.toContainAnyEntries(expected)

Expected object to not contain any of the provided entries:
  [[\\"a\\", \\"qux\\"], [\\"a\\", \\"foo\\"]]
Received:
  {\\"a\\": \\"foo\\", \\"b\\": \\"bar\\", \\"c\\": \\"baz\\"}"
`;

exports[`.toContainAnyEntries fails when given object does not contain any of the given entries 1`] = `
"expect(received).toContainAnyEntries(expected)

Expected object to contain any of the provided entries:
  [[\\"a\\", \\"qux\\"], [\\"b\\", \\"foo\\"]]
Received:
  {\\"a\\": \\"foo\\", \\"b\\": \\"bar\\", \\"c\\": \\"baz\\"}"
`;
